Add Windows support

benmoss opened the pull request:
**What this PR does / why we need it**:
Adds the releases required to deploy Windows nodes to clusters with flannel/host-gw networking.

- Adds windows versions of releases
- Adds start script for windows (in go; did not include vendor directory due to so many deps)

**How can this PR be verified?**
Deploy a cluster with the changes to kubo-deployment.

**Is there any change in kubo-deployment?**
Yes.

**Is there any change in kubo-ci?**
No.

**Does this affect upgrade, or is there any migration required?**
We added a node-selector to the linux specs, to avoid scheduling them on windows nodes. This would apply on linux clusters as well, but not actually change anything.

**Which issue(s) this PR fixes:**
Lack of windows support. :)

**Release note**:
<!--
Write your release note:
1. Enter your extended release note in the below block. If the PR requires
   additional action from users switching to the new release, include the
   string "action required".
3. If no release note is required, just write "NONE".
-->
```release-note
Add support for windows nodes
```

astrieanna commented:
https://github.com/cloudfoundry-incubator/kubo-deployment/pull/354

alex-slynko commented on .final_builds/packages/golang-1.10-windows/index.yml:
```diff
@@ -0,0 +1,6 @@
+builds:
```
Any particular reason you use Go 1.10?

benmoss replied:
I think when we started on our work CFCR was still on 1.10 as well, we'll update to match

benmoss commented:
The binaries story is a little complicated, as we're relying on one unmerged PR and one unreleased change.

- For the CNI plugins they should be released with all the others [here](https://github.com/containernetworking/plugins/). There hasn't been a release since the Windows ones were merged in yet, maybe we can poke someone to get a release made.
- For Flannel we built our binary from [a PR](https://github.com/coreos/flannel/pull/921) that is still not merged.
- For Kubernetes we got our binaries from the [standard releases](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.11.md#node-binaries-1), they're in kubernetes-node-windows-amd64.tar.gz.

benmoss commented:
We've rebased against develop and changed the merge target to develop. We made the hostname change you mentioned as well. We'd like to chat and maybe x-team pair on the args changes, and we were thinking about deferring the Azure support if that'd be okay with you all.

benmoss commented:
@alex-slynko and I chatted about this and we agreed to split this up into a smaller first iteration that comes along with some test coverage added to https://github.com/cloudfoundry-incubator/kubo-ci
